python/delta/tests/test_deltatable.py

```diff
@@ -791,6 +791,34 @@ def test_protocolUpgrade(self) -> None:
         with self.assertRaisesRegex(ValueError, "writerVersion"):
             dt.upgradeTableProtocol(1, {})  # type: ignore[arg-type]
 
+    def test_restoreToVersion(self) -> None:
+        self.__writeDeltaTable([('a', 1), ('b', 2)])
+        self.__overwriteDeltaTable([('a', 3), ('b', 2)],
+                                   schema=["key_new", "value_new"],
+                                   overwriteSchema='true')
+        DeltaTable.forPath(self.spark, self.tempFile).restoreToVersion(0)
+        restored = DeltaTable.forPath(self.spark, self.tempFile).toDF()
+
+        self.__checkAnswer(restored, [Row(key='a', value=1), Row(key='b', value=2)])
+
+    def test_restoreToTimestamp(self) -> None:
+        self.__writeDeltaTable([('a', 1), ('b', 2)])
+        timestampToRestore = DeltaTable.forPath(self.spark, self.tempFile) \
+            .history() \
+            .head() \
+            .timestamp \
+            .strftime('%Y-%m-%d %H:%M:%S.%f')
+
+        self.__overwriteDeltaTable([('a', 3), ('b', 2)],
+                                   schema=["key_new", "value_new"],
+                                   overwriteSchema='true')
```

Comment:
sorry its not clean on why do we need to specify the new schema and overwrite it. Isn't it the same schema as the v0?

Reply:
`__overwriteDeltaTable` overwrites table with a new schema and new data. This helps us to show that restore% methods restores schema and data.
Some code examples to show this in details:
```python
df = spark.createDataFrame([(1, 'a')], ["key", "value"])
df.write.format("delta").mode("overwrite").save("/tmp/delta_test")
spark.read.format("delta").load("/tmp/delta_test").printSchema()
root
 |-- key: long (nullable = true)
 |-- value: string (nullable = true)

df = spark.createDataFrame([(1, 'a')], ["key_new", "value_new"])
df.write.format("delta").mode("overwrite").option('overwriteSchema', 'true').save("/tmp/delta_test")
spark.read.format("delta").load("/tmp/delta_test").printSchema()
root
 |-- key_new: long (nullable = true)
 |-- value_new: string (nullable = true)
```

Reply:
Ah.. I see. that makes sense. Thanks for the explanation.